tool/tsh/common/tsh.go

```diff
@@ -356,6 +356,17 @@ func onSSH(cf *CLIConf) {
 		utils.FatalError(err)
 	}
 
+	// set the OnAuthentication callback which gets called with the authentication
+	// result. if the user has successfully authenticated, save their profile.
+	tc.OnAuthentication = func(ok bool) {
+		if ok {
+			err := tc.SaveProfile("")
+			if err != nil {
+				utils.FatalError(err)
+			}
+		}
+	}
```

Comment:
Looks like the other call to SaveProfile() needs to be removed because this one should cover all cases.